packages/components/message/src/Message.comp.tsx

```diff
@@ -0,0 +1,116 @@
+import { defineComponent, onBeforeUnmount, onMounted, SetupContext, isVNode, h, computed } from 'vue'
+import IxIcon from '@idux/components/icon/src/Icon.vue'
+import { PropTypes, isUndefined } from '@idux/cdk/utils'
+import { MessageItemProps, useGlobalConfig } from '@idux/components'
+import { Subject } from '@idux/cdk/subject'
+import { MessageType } from './types'
+
+export default defineComponent({
+  name: 'IxMessageItem',
+  components: {
+    IxIcon,
+  },
+  props: {
+    duration: PropTypes.number,
+    pauseOnHover: PropTypes.bool,
+    icon: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).def(''),
+    content: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).def(''),
+    type: PropTypes.oneOfType([PropTypes.oneOf(['success', 'error', 'info', 'warn'] as const), PropTypes.string]).def(
+      '',
+    ),
+    messageId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  },
+  emits: ['close'],
+  setup(props: MessageItemProps, { emit }: SetupContext) {
+    // init
+    const classes = useClasses(props.type)
+    const messageConfig = useGlobalConfig('message')
+    const id = props.messageId
+    const { closeSubject, startCloseTimer, clearCloseTimer, destroy } = usePauseOnHover({
+      duration: props.duration ?? messageConfig.duration,
+      pauseOnHover: props.pauseOnHover ?? messageConfig.pauseOnHover,
+    })
+    closeSubject.subscribe(() => emit('close', id))
+    onMounted(() => {
+      startCloseTimer()
+    })
+    onBeforeUnmount(() => {
+      destroy()
+    })
+
+    return {
+      classes,
+      startCloseTimer,
+      clearCloseTimer,
+    }
+  },
+  render() {
+    const content = isVNode(this.content) ? h(this.content) : this.content
+    const icon = isVNode(this.icon) ? h(this.icon) : getIcon(this.icon as string, this.type)
+
+    return (
+      <div class={this.classes} onMouseenter={this.clearCloseTimer} onMouseleave={this.startCloseTimer}>
+        <div class="ix-message-item__content">
```

Comment:
```suggestion
        <div class="ix-message-item-content">
```
class 命名不使用 下划线，统一使用中划线。